### PR TITLE
Replace PathUtil.Combine and GetExtension with Path.

### PR DESCRIPTION
**Pull request by gerhardol**
## Proposed changes

Prepare to replace PathUtil.Combine and PathUtil.GetExtension with the .NET5 Path variants directly. 
https://docs.microsoft.com/dotnet/core/compatibility/2.1#path-apis-dont-throw-an-exception-for-invalid-characters

.NET5 Path no longer throws ArgumentException on
"Illegal characters in path".
Replace PathUtil Combine and GetExtinsion with the Path
variant directly as well removing other catches of ArgumentException.

* Remove .NET5 todo

## Test methodology 

Tests were updated - when PathUtil versions are removed the tests will be removed though

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by mstv — COMMENTED** [review]
LGTM if the `Path` methods would not throw. But the documentation mentions exceptions on invalid characters.

**Comment by gerhardol** [comment]
> LGTM if the `Path` methods would not throw. But the documentation mentions exceptions on invalid characters.

The web doc is OK: https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getextension?view=net-5.0

The metadata documentation is incorrect. So was the online configuration until recently.
RussKie pointed to a blog that described the changes (there was a GitHub issue too).
A quite big change to not be documented.

File: GitCommands/PathUtil.cs
Comment (RussKie) [inline GitCommands/PathUtil.cs:158]: Suggest removing this method completely

**Review by RussKie — COMMENTED** [review]
👍

File: GitCommands/PathUtil.cs
Comment (gerhardol) [reply GitCommands/PathUtil.cs:158]: PathUtil.Combine will be removed too

I wanted to have this approved and some other PRs merged before I made that change
So I will rebase and make the change and change this from draft (and then quickly merge after approval, assuming that opinions are raised on the draft...)

**Comment by gerhardol** [comment]
Rebased and removed the replaced PathUtil methods
Will of course need a squash before merge
Ready for final reviews

**Review by RussKie — APPROVED** [review]

**Comment by gerhardol** [comment]
Rebased and squashed without changes, adjusted commit message.

**Comment by gerhardol** [comment]
@msftbot squash in 24 hours

**Comment by gerhardol** [comment]
@msftbot merge in 24 hours

File: GitUI/CommandsDialogs/FormResolveConflicts.cs
Comment (mstv) [inline GitUI/CommandsDialogs/FormResolveConflicts.cs:331]:
```suggestion
                string? extension = Path.GetExtension(fileName)?.ToLower();
```

File: Plugins/Statistics/GitStatistics/LineCounter.cs
Comment (mstv) [inline Plugins/Statistics/GitStatistics/LineCounter.cs:57]: Does `HashSet<string>.Contains(null)` thriow?

Comment (gerhardol) [reply Plugins/Statistics/GitStatistics/LineCounter.cs:57]: I would assume so - this is a potential issue
(not changed by this PR)

Comment (gerhardol) [reply Plugins/Statistics/GitStatistics/LineCounter.cs]: I moved the change into the try-catch

File: GitUI/CommandsDialogs/FormResolveConflicts.cs
Comment (gerhardol) [reply GitUI/CommandsDialogs/FormResolveConflicts.cs]: ok

File: Plugins/Statistics/GitStatistics/LineCounter.cs
Comment (mstv) [reply Plugins/Statistics/GitStatistics/LineCounter.cs]: This has been changed by this PR: `PathUtil.GetExtension` did not return `null` but `""`. So
```suggestion
                    if (extensions.Contains(Path.GetExtension(file) ?? ""))
```
?

**Review by mstv — APPROVED** [review]
👍

**Pull request merged**
